### PR TITLE
Mention 2 cpp-reviewer requirement in pull request template

### PR DESCRIPTION
**Pull request by davidwendt**
Although 2 cpp reviewers are automatically assigned to a PR, there is no documentation that indicates 2 are required. This PR adds to the list of guidelines in the [Pull Request Template](https://github.com/rapidsai/cudf/blob/branch-22.06/.github/PULL_REQUEST_TEMPLATE.md) that appears when a PR is created through the github web page.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (wence-) [inline .github/PULL_REQUEST_TEMPLATE.md:51]:
```suggestion
   will automatically be assigned two cudf-cpp-codeowners reviewers.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (wence-) [inline .github/PULL_REQUEST_TEMPLATE.md:52]: Is this something that the submitter needs to worry about, or is it rather the responsibility of the eventual merger?

Comment (davidwendt) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: Both I think.

Comment (wence-) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: I guess it's part of the PR workflow, however, it's easy to forget the instructions in the template (since I guess one deletes it and replaces it with the PR summary). At least for me, in the github web UI it's rather hard to see which approving reviews would match this criterion. e.g. looking at https://github.com/rapidsai/cudf/pull/10761 one has to mouse over the individual usernames to see which groups they belong to (and then figure out if the criteria are met). WDYT?

I wonder if the automerging functionality of @rapids-ops-bot can be augmented to apply the criteria, and then this instruction would say something like: "Pull requests that modify cpp source require at least two approvals from cudf-cpp-codeowners before merge."

Comment (davidwendt) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: The assignments are automatic although somewhat random. So you will have 2 cpp reviewers assigned if you modify cpp files. You are on your honor to **not ignore** these and to merge with 2 approvals. This is not something that can be enforced through the merger at this time so I felt it should be at least documented in this template.

Along with the mouse hover, you can see was assigned in the PR stream. For example here is where the assignments occurred in #10761 
image

Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: Because this rule is specific to C++ developers and not all of the repository, I would recommend putting this in the [C++ Developer Guide](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/docs/DEVELOPER_GUIDE.md). I share the concern that the PR template is likely to be missed / deleted.

Additionally, the requirement of two C++ reviews does not affect the actions of an external contributor who is submitting a PR since a RAPIDS developer will be the one responsible for review and merge. I view external contributors as the primary readers of the PR template, since experienced developers on the RAPIDS team probably pay it no attention.

Eventually I would like to minimize the long comments in this PR template and replace it with a checklist style of PR template that gives all contributors actionable items. Examples: 
- https://axolo.co/blog/p/part-3-github-pull-request-template
- https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/
- https://docs.microsoft.com/en-us/azure/devops/repos/git/pull-request-templates

Comment (wence-) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: Thanks, makes sense given the current merger restrictions.

**Review by bdice — APPROVED** [review]
LGTM. This will be helpful at least until #10774 and a second PR I'm planning to improve the CONTRIBUTING guide are complete.

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**
